### PR TITLE
Run3-sim54 First step to adopt Ecal Reco geometry for DD4Hep

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

First step to adopt ECAL reco geometry for DD4Hep

#### PR validation:

Use standard runTheMatrix.py

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/CaloEventSetup/interface/CaloGeometryLoader.icc [hunk 1]
```diff
@@ -98,17 +102,78 @@ void CaloGeometryLoader<T, D>::makeGeometry(const D* cpv,
...
+    }
+    ++counter;
+
+    const ParmVec& parameters(fv.parameters());
+
+    DD3Vector x, y, z;
+    fv.rotation().GetComponents(x, y, z);
+    const CLHEP::HepRep3x3 temp(x.X(), y.X(), z.X(), x.Y(), y.Y(), z.Y(), x.Z(), y.Z(), z.Z());
+    const CLHEP::HepRotation hr(temp);
+    const CLHEP::Hep3Vector h3v(fv.translation().X(), fv.translation().Y(), fv.translation().Z());
```
Comment (cvuosalo) [inline Geometry/CaloEventSetup/interface/CaloGeometryLoader.icc]: We would like to replace CLHEP matrices and vectors with their ROOT equivalents. New code should use the ROOT versions. See: https://hypernews.cern.ch/HyperNews/CMS/get/swDevelopment/3549.html

Comment (bsunanda) [reply Geometry/CaloEventSetup/interface/CaloGeometryLoader.icc]: Yes I understand that. We shall do that in one go. We have to handle several classes in CaloGeometry, Ecal, Hcal, HGCal geometry at the same time. I have it on my to-do list. I shall do that later

**Comment by cvuosalo** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**